Plugin install/delete test & fixes

oguzkocer opened the pull request:
This PR changes a few things that should hopefully come close to finalizing the Plugin Management on FluxC:

1. Use v1.2 for all endpoints so the response json is consistent (due to the naming change)
2. Add a few custom error types for installation (pending verification from server side)
3. Adds a test that will fetch all plugins on a site, remove the BuddyPress plugin if it exists, install it and delete it. I've added a comment why I opted to add this test in this way. Let me know if it's not clear enough

In order to test:
* The test should be done using the beta jetpack site credentials, however it might not always work. The fetch v1.2 is a bit delayed for some reason and which is currently being investigated. However, this only affects the first step of the test and as long as you can uninstall the plugin before testing, it should(!) work everytime since we install and then delete the plugin leaving it in the same state.

Note that we might need to make some small changes or add more tests soon, but I think it's ready to be merged to develop. Once this PR is merged, I'll open a PR to merge the master branch to be merged into `develop` and then use the hash to update the WPAndroid side.

/cc @tonyr59h

enejb commented on example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_PluginTestJetpack.java:
```diff
@@ -109,6 +98,31 @@ public void testUpdateSitePlugin() throws InterruptedException {
+            }
+        }
+
+        // Install the Buddypress plugin
```
I would install a smaller plugin. BuddyPress is huge (2.8MB) and it if you are running the test locally it would take a long time. I would suggest installing react https://en-ca.wordpress.org/plugins/react/ which is only 83kb.

oguzkocer replied:
I didn't even think about that, thanks @enejb, I'll make the change! /cc @tonyr59h

oguzkocer commented:
@tonyr59h I've added a DB check to make sure the plugin is actually deleted which revealed a small issue I've fixed in 463d64a83e95197cc54203b2264983a757f7a92b. It should be ready for review now, but I am unable to fully test it right now (waiting for the fetch v1.2 to work correctly as it's returning the cached version right now).

tonyr59h commented on example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_PluginTestJetpack.java:
```diff
@@ -109,6 +98,36 @@ public void testUpdateSitePlugin() throws InterruptedException {
     }
 
+    // It's both easier and more efficient to combine install & delete tests since we need to make sure we have the
+    // plugin installed for the delete test and the plugin is not installed for the install test
```
Thanks for adding the comments.
